mkdocs.yml

```diff
@@ -0,0 +1,116 @@
+site_name: Goss
+site_description: Goss is a YAML based serverspec alternative tool for validating a server’s configuration.
+site_author: Goss team
+site_url: https://goss.readthedocs.io/
+repo_url: https://github.com/goss-org/goss
+repo_name: goss-org/goss
+edit_uri: edit/master/docs/
+
+
+theme:
+  name: material
+  palette:
+    - media: "(prefers-color-scheme: light)"
+      scheme: default
+      primary: black
+      accent: amber
+      toggle:
+        icon: material/weather-sunny
+        name: Switch to dark mode
+    - media: "(prefers-color-scheme: dark)"
+      scheme: slate
+      primary: black
+      accent: indigo
+      toggle:
+        icon: material/weather-night
+        name: Switch to light mode
+  features:
+    - content.action.edit
+    - content.code.copy
+    - navigation.footer
+    - navigation.instant
+    - navigation.instant.progress
+    - navigation.top
+    - navigation.tracking
+    - search.highlight
+    - search.share
+    - search.suggest
+    - toc.follow
+
+
+extra_css:
+  - style.css
+
+plugins:
+  - search
+  - macros:
+      render_by_default: false
+
+markdown_extensions:
+  - abbr
+  - admonition
+  - attr_list
+  - def_list
+  - md_in_html
+  - mdx_breakless_lists
+  - tables
+  - pymdownx.details
+  - pymdownx.emoji:
+      emoji_index: !!python/name:material.extensions.emoji.twemoji
+      emoji_generator: !!python/name:material.extensions.emoji.to_svg
+  - pymdownx.highlight:
+      anchor_linenums: true
+      line_spans: __span
+      pygments_lang_class: true
+  - pymdownx.inlinehilite
+  - pymdownx.magiclink:
+      repo_url_shortener: true
+      social_url_shortener: true
+      repo_url_shorthand: true
+      social_url_shorthand: true
+      user: goss-org
+      repo: goss
+  - pymdownx.snippets:
+      base_path:
+        - .
+        - docs/snippets
+      check_paths: true
+  - pymdownx.superfences
+
+
+nav:
```

Comment:
In previous sites I've used https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin to allow the auto generated navigation to be used, and where it wasn't what we wanted, allowed us to hint it via a `.pages` in such directories. That worked well and let us remove friction caused by needing to maintain this section.

(I can do as follow-up, or stack to this if you like?)

Reply:
I never used it, so I might need some time to take a look (and I will).
I would advise finishing this review and merge it as it is (in term of tooling), then iterate with extra PR to bring or modify tooling for more comfort.
In the case of this extension, I am not comfortable to be the one to submit it in the PR given I never used it, and so I don't have a legit experience to propose it with a proper setting and workflow

Reply:
All good; I can follow up.

Reply:
Finally I took the time to look at it and integrate it 👍🏼 
Navigation is now handled in `.pages` file susing `mkdocs-awesome-pages-plugin`